frontend/src/features/Train/components/ImageUploadLayout.tsx

```diff
@@ -0,0 +1,207 @@
+  const mime = matched ? matched[1] : undefined;
+  const bstr = atob(arr[1]);
+  let n = bstr.length;
+  const u8arr = new Uint8Array(n);
+  while (n) {
+    u8arr[n - 1] = bstr.charCodeAt(n - 1);
+    n -= 1; // to make eslint happy
+  }
+  return new File([u8arr], filename, { type: mime });
+};
+
+export const UploadImagePanel = ({
+  dataSource,
+  methods,
+}: {
+  dataSource: DATA_SOURCE;
+  methods: UseFormReturn<ImageUploadData, unknown>;
+}) => {
+  const [uploadFile] = useUploadDatasetFileMutation();
+  const { data, refetch } = useGetDatasetFilesDataQuery({ dataSource });
+  return (
+    <>
+      {methods.formState.errors.name && (
+        <Typography>Please select a file</Typography>
+      )}
+      <Stack direction={"row"} spacing={2}>
+        <FilerobotImageEditor
+          source="https://scaleflex.airstore.io/demo/stephen-walker-unsplash.jpg"
+          onSave={(editedImageObject: unknown) => {
+            const file = dataURLtoFile(
+              editedImageObject.imageBase64,
+              editedImageObject.fullName
+            );
+            uploadFile({ dataSource, file });
+          }}
+          annotationsCommon={{
+            fill: "#ff0000",
+          }}
+          Text={{ text: "Filerobot..." }}
+          Rotate={{ angle: 90, componentType: "slider" }}
+          Crop={{
+            presetsItems: [
+              {
+                titleKey: "classicTv",
+                descriptionKey: "4:3",
+                ratio: (4 / 3).toString(),
+              },
+              {
+                titleKey: "cinemascope",
+                descriptionKey: "21:9",
+                ratio: (21 / 9).toString(),
+              },
+            ],
+            presetsFolders: [
+              {
+                titleKey: "socialMedia",
+
+                // icon: Social, // optional, Social is a React Function component. Possible (React Function component, string or HTML Element)
+                groups: [
+                  {
+                    titleKey: "facebook",
+                    items: [
+                      {
+                        titleKey: "profile",
+                        width: 180,
+                        height: 180,
+                        descriptionKey: "fbProfileSize",
+                      },
+                      {
+                        titleKey: "coverPhoto",
+                        width: 820,
+                        height: 312,
+                        descriptionKey: "fbCoverPhotoSize",
+                      },
+                    ],
+                  },
+                ],
+              },
+            ],
+          }}
+          tabsIds={[TABS.ADJUST, TABS.ANNOTATE, TABS.WATERMARK]} // or {['Adjust', 'Annotate', 'Watermark']}
+          defaultTabId={TABS.ANNOTATE} // or 'Annotate'
+          defaultToolId={TOOLS.TEXT} // or 'Text'
+          savingPixelRatio={0}
+          previewPixelRatio={0}
+        />
+        <Button variant="outlined" onClick={() => refetch()}>
```

Comment:
Refresh button should be positioned so that it is below the image editor instead of to the right of it